mlflow/types/schema.py

```diff
@@ -408,3 +408,151 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: str,  # pylint: disable=redefined-builtin
+        default: Any = None,
+        optional: bool = True,
+    ):
+        self._name = str(name)
+        self._type = str(type)
+        self._default = default
+        self._optional = optional
+        self._check_default_type()
+
+    def _check_default_type(self):
+        if self.default is not None and type(self.default).__name__ != self.type:
```

Comment:
If `self.default` is a string, what is the expected value of `self.type`? `str`?

Reply:
Yes